core/graphelier-service/db/mongo.go

```diff
@@ -334,54 +385,71 @@ func (c *Connector) GetTopOfBookByInterval(instrument string, startTimestamp uin
 	if err != nil {
 		return nil, err
 	}
-	log.Tracef("Documents in interval {%d,%d}=%d, keeping 1 in %d", startTimestamp/interval, endTimestamp/interval, count, count/maxCount)
 
-	// Add $mod comparator to interval_multiple filter
-	filter[1].Value = append(filter[1].Value.(bson.D), bson.E{Key: "$mod", Value: bson.A{count / maxCount, 0}})
+	// if maxCount < count, find points with big interval (mongo query)
+	// else find points with small interval
+	if maxCount < count {
+		log.Tracef("Documents in orderbook interval {%d,%d}=%d, keeping 1 in %d", startTimestamp/interval, endTimestamp/interval, count, count/maxCount)
+		// Add $mod comparator to interval_multiple filter
+		filter[1].Value = append(filter[1].Value.(bson.D), bson.E{Key: "$mod", Value: bson.A{count / maxCount, 0}})
 
-	// Project snapshots to keep only the best bid and ask
-	findOptions := options.Find()
-	findOptions.Projection = bson.D{
-		{Key: "interval_multiple", Value: 1},
-		{Key: "bids", Value: bson.D{{Key: "$slice", Value: 1}}},
-		{Key: "asks", Value: bson.D{{Key: "$slice", Value: 1}}},
-		{Key: "bids.price", Value: 1},
-		{Key: "asks.price", Value: 1},
+		results, err = getBigTopBookInterval(&filter, collection, interval)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		orderbook, err := c.GetOrderbook(instrument, startTimestamp)
+		if err != nil {
+			return nil, err
+		}
+		messages, err := c.GetMessagesWithinInterval(instrument, startTimestamp, endTimestamp)
+		if err != nil {
+			return nil, err
+		}
+		x := len(messages) / int(maxCount)
```

Comment:
Shouldn't you be using the timestamp range here to determine how long your interval is?
```suggestion
		x := (endTimestamp - startTimestamp) / maxCount
```